docs-site/content/guide/high-availability.md

```diff
@@ -272,3 +272,137 @@ final config = Configuration(
 
   </template>
 </Tabs>
+ 
+ ## Deploy a Typesense cluster to Docker Swarm  
+
+To Deploy a Typesense cluster to Docker Swarm, follow these steps:
+
+- Initialize Docker Swarm. For example, in a 4 node setup, target docker engine on one of the nodes to act as a manager
+
+```shell  
+docker swarm init --advertise-addr $(hostname -i)
+Swarm initialized: current node (6082x127zz98f0pwgjexbv5xp) is now a manager.
+   
+To add a worker to this swarm, run the following command:
+
+docker swarm join --token SWMTKN-1-30txqn35hmpwjpk2qq2zmled1rf94pcft2nbhsb0ckleco9pb2-bjh6oh9yz3vk58uimd6v3jjky 192.168.0.25:2377
+
+To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+```
+
+- Add docker instance on rest of the nodes to existing Swarm as node. Repeat the below command on all nodes
+
+```shell
+docker swarm join --token SWMTKN-1-30txqn35hmpwjpk2qq2zmled1rf94pcft2nbhsb0ckleco9pb2-bjh6oh9yz3vk58uimd6v3jjky 192.168.0.25:2377
+This node joined a swarm as a worker.
+
+```
+
+- Create a new nodes file on each node that's part of the Swarm 
+
+```shell
+mkdir typesense && cd typesense && echo 'typesense-1:8107:8108,typesense-2:7107:7108,typesense-3:9107:9108' | sudo tee nodes
+```
+
+- Create data folder on all the nodes
+
+ ```shell
+ mkdir /tmp/typesense-data-1/ &&  mkdir /tmp/typesense-data-2/ && mkdir /tmp/typesense-data-3/
```

Comment:
Shouldn't this be run on separate nodes, one at a time?

Reply:
can be run in one go

Reply:
if the service moves to other node the data folder should be present

Reply:
The data folder needs to be present, typesense doesn't auto-create it.